src/libraries/System.Private.CoreLib/src/System/Math.cs

```diff
@@ -1682,5 +1682,157 @@ internal static ulong ConvertToUInt64Checked(double value)
             ThrowHelper.ThrowOverflowException();
             return 0;
         }
+
+        [StackTraceHidden]
+        internal static int DivideChecked(int dividend, int divisor)
+        {
+            if ((uint)(divisor + 1) <= 1)
+            {
+                if (divisor == 0)
+                {
+                    ThrowHelper.ThrowDivideByZeroException();
+                }
+                else if (divisor == -1)
+                {
+                    if (dividend == int.MinValue)
+                    {
+                        ThrowHelper.ThrowOverflowException();
+                    }
+                    return -dividend;
+                }
+            }
+            return dividend / divisor;
```

Comment:
I do not think that this will work. I expect that this will hit infinite recursion on arm - the only platform where these are used.

This needs to leave the actual division to the unmanaged code.